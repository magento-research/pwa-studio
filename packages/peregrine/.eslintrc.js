const config = {
    parser: 'babel-eslint',
    extends: ['@magento'],
    rules: {
        "no-undef": "off"
    }
};

module.exports = config;
