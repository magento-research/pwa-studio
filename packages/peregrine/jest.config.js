module.exports = {
    displayName: 'Peregrine',
    setupFiles: ['<rootDir>/scripts/shim.js', '<rootDir>/scripts/fetch-mock.js']
};
