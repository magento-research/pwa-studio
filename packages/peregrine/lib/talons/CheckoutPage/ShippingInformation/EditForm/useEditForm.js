import { useCallback } from 'react';
import { useMutation } from '@apollo/react-hooks';

import { useCartContext } from '../../../../context/cart';

export const useEditForm = props => {
    const {
        mutations: { setShippingInformationMutation },
        shippingData
    } = props;

    const [{ cartId }] = useCartContext();
    const [setShippingInformation, { called, loading }] = useMutation(
        setShippingInformationMutation
    );

    const { country, region } = shippingData;
    const { code: countryCode } = country;
    const { code: regionCode } = region;

    const initialValues = {
        ...shippingData,
        country: countryCode,
        region: regionCode
    };

    const handleSubmit = useCallback(
        formValues => {
            const { country, email, ...address } = formValues;
            setShippingInformation({
                variables: {
                    cartId,
                    email,
                    address: {
                        ...address,
                        country_code: country
                    }
                }
            });
        },
        [cartId, setShippingInformation]
    );

    return { handleSubmit, initialValues, isSaving: called && loading };
};
