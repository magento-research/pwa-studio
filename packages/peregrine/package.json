{
  "name": "@magento/peregrine",
  "version": "3.0.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "The core runtime of Magento PWA",
  "main": "lib/index.js",
  "scripts": {
    "build": " ",
    "build:ci": " ",
    "build:cjs": " ",
    "build:dev": " ",
    "clean": " ",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "storybook:build": "build-storybook -c .storybook -o storybook-dist",
    "watch": " "
  },
  "repository": "github:magento-research/pwa-studio",
  "author": "Magento Commerce",
  "license": "(OSL-3.0 OR AFL-3.0)",
  "bugs": {
    "url": "https://github.com/magento-research/pwa-studio/issues"
  },
  "homepage": "https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine#readme",
  "devDependencies": {
    "@storybook/react": "~4.1.13",
    "intl": "~1.2.5",
    "react": "~16.8.3",
    "react-apollo": "~2.5.3",
    "react-dom": "~16.8.3",
    "react-router-dom": "~5.0.0",
    "react-test-renderer": "~16.8.6",
    "rimraf": "~2.6.3"
  },
  "peerDependencies": {
    "@babel/runtime": "^7.4.2",
    "react": "^16.8.0",
    "react-apollo": "^2.5.3",
    "react-dom": "^16.8.0",
    "react-router-dom": "~5.0.0"
  },
  "engines": {
    "node": "8.10.0 || >=10.7.0",
    "yarn": ">=1.12.0"
  },
  "module": "lib/index.js",
  "jsnext:main": "lib/index.js",
  "es2015": "lib/index.js",
  "sideEffects": false
}
