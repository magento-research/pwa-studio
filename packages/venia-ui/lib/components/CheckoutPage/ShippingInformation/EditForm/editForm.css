.root {
    display: grid;
    gap: 0.5rem 1.5rem;
    grid-template-columns: 1fr 1fr;
    width: 100%;
}

.field {
    grid-column: 1 / span 2;
}

.firstname {
    grid-column: 1;
}

.lastname {
    grid-column: 2;
}

.submit {
    composes: root_normalPriority from '../../../Button/button.css';
    font-size: 0.875rem;
    font-weight: 600;
    grid-column: 1 / span 2;
    justify-self: center;
    margin-top: 1rem;
}

@media (max-width: 960px) {
    .firstname {
        grid-column: 1 / span 2;
    }

    .lastname {
        grid-column: 1 / span 2;
    }
}
