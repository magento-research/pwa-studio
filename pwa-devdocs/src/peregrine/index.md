---
title: Peregrine
---

The Peregrine project is a collection of UI Components for Magento PWA projects.
Use, extend, or remix these components to create a unique Magento PWA storefront.

Visit the [`Peregrine`] repository to contribute to this project.

[`Peregrine`]: https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine
