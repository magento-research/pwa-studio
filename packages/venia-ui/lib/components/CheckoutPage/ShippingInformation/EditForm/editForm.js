import React from 'react';
import { Form } from 'informed';
import { useEditForm } from '@magento/peregrine/lib/talons/CheckoutPage/ShippingInformation/EditForm/useEditForm';

import { mergeClasses } from '../../../../classify';
import { isRequired } from '../../../../util/formValidators';
import Button from '../../../Button';
import Country from '../../../Country';
import Field, { Message } from '../../../Field';
import Region from '../../../Region';
import TextInput from '../../../TextInput';
import defaultClasses from './editForm.css';
import EditFormOperations from './editForm.gql';

const EditForm = props => {
    const { afterSubmit, classes: propClasses, shippingData } = props;
    const talonProps = useEditForm({
        afterSubmit,
        ...EditFormOperations,
        shippingData
    });
    const { handleSubmit, initialValues, isSaving, isUpdate } = talonProps;
    const classes = mergeClasses(defaultClasses, propClasses);

    return (
        <Form
            className={classes.root}
            initialValues={initialValues}
            onSubmit={handleSubmit}
        >
            <Field classes={{ root: classes.field }} id="email" label="Email">
                <TextInput field="email" validate={isRequired} />
                <Message fieldState={{}}>
                    Set a password at the end of guest checkout to create an
                    account in one easy step.
                </Message>
            </Field>
            <Field
                classes={{ root: classes.firstname }}
                id="firstname"
                label="First Name"
            >
                <TextInput field="firstname" validate={isRequired} />
            </Field>
            <Field
                classes={{ root: classes.lastname }}
                id="lastname"
                label="Last Name"
            >
                <TextInput field="lastname" validate={isRequired} />
            </Field>
            <Country classes={{ root: classes.field }} validate={isRequired} />
            <Field
                classes={{ root: classes.field }}
                id="street0"
                label="Street Address"
            >
                <TextInput field="street[0]" validate={isRequired} />
            </Field>
            <Field
                classes={{ root: classes.field }}
                id="street1"
                label="Street Address 2"
            >
                <TextInput field="street[1]" />
            </Field>
            <Field classes={{ root: classes.field }} id="city" label="City">
                <TextInput field="city" validate={isRequired} />
            </Field>
            <Region classes={{ root: classes.field }} validate={isRequired} />
            <Field
                classes={{ root: classes.field }}
                id="postcode"
                label="ZIP / Postal Code"
            >
                <TextInput field="postcode" validate={isRequired} />
            </Field>
            <Field
                classes={{ root: classes.field }}
                id="telephone"
                label="Phone Number"
            >
                <TextInput field="telephone" validate={isRequired} />
            </Field>
            <Button
                classes={{ root_normalPriority: classes.submit }}
                disabled={isSaving}
                priority="normal"
                type="submit"
            >
                {isUpdate ? 'Update' : 'Continue to Shipping Method'}
            </Button>
        </Form>
    );
};

EditForm.defaultProps = {
    shippingData: {
        country: {
            code: 'US'
        },
        region: {
            code: ''
        }
    }
};

export default EditForm;
