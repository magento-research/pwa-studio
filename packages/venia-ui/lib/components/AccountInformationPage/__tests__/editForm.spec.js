import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import EditForm from '../editForm';

jest.mock('../../../classify');
jest.mock('../../FormError', () => 'FormError');

const onCancelModal = jest.fn().mockName('onCancelModal');
const handleSubmit = jest.fn().mockName('handleSubmit');
const handleChangePassword = jest.fn().mockName('handleChangePassword');
const mockProps = {
    formErrors: new Map(),
    onCancelModal,
    handleSubmit,
    handleChangePassword,
    activeChangePassword: false,
    isDisabled: false,
    informationData: {
        firstname: 'Huy',
        lastname: 'Kon',
        email: 'huykon@gmail.com'
    }
};

test('renders empty form without data', () => {
    const tree = createTestInstance(<EditForm {...mockProps} />);
    expect(tree.toJSON()).toMatchSnapshot();
});

test('renders form error', () => {
    const formErrors = new Map([['error', new Error('Form Error')]]);

    const tree = createTestInstance(
        <EditForm {...mockProps} formErrors={formErrors} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
});

describe('renders prefilled form with data', () => {
    const informationData = {
        email: 'fry@planet.express',
        firstname: 'Philip',
        lastname: 'Fry'
    };

    test('with enabled change password form', () => {
        const newProps = {
            ...mockProps,
            informationData,
            activeChangePassword: true
        };

        const tree = createTestInstance(<EditForm {...newProps} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('with disabled buttons', () => {
        const newProps = { ...mockProps, informationData, isDisabled: true };

        const tree = createTestInstance(<EditForm {...newProps} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
