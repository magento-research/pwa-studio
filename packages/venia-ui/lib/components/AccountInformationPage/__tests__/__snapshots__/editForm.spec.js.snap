// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty form without data 1`] = `
Array [
  <FormError
    errors={Map {}}
  />,
  <form
    className="root"
    onKeyDown={[Function]}
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="firstname"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="firstname"
        >
          First Name
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              name="firstname"
              onBlur={[Function]}
              onChange={[Function]}
              value="Huy"
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="lastname"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="lastname"
        >
          Last Name
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              name="lastname"
              onBlur={[Function]}
              onChange={[Function]}
              value="Kon"
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="email"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="email"
        >
          Email
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              value="huykon@gmail.com"
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="passwordLabel"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="password"
        >
          Password
        </label>
        <span>
          **********
        </span>
      </div>
    </div>
    <div
      className="changePasswordButtonContainer"
    >
      <button
        className="root_normalPriority"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="content"
        >
          Change Password
        </span>
      </button>
    </div>
    <div
      className="buttons"
    >
      <button
        className="root_lowPriority"
        disabled={false}
        onClick={[MockFunction onCancelModal]}
        type="button"
      >
        <span
          className="content"
        >
          Cancel
        </span>
      </button>
      <button
        className="root_highPriority"
        disabled={false}
        type="submit"
      >
        <span
          className="content"
        >
          Save
        </span>
      </button>
    </div>
  </form>,
]
`;

exports[`renders form error 1`] = `
Array [
  <FormError
    errors={
      Map {
        "error" => [Error: Form Error],
      }
    }
  />,
  <form
    className="root"
    onKeyDown={[Function]}
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="firstname"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="firstname"
        >
          First Name
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              name="firstname"
              onBlur={[Function]}
              onChange={[Function]}
              value="Huy"
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="lastname"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="lastname"
        >
          Last Name
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              name="lastname"
              onBlur={[Function]}
              onChange={[Function]}
              value="Kon"
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="email"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="email"
        >
          Email
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              value="huykon@gmail.com"
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="passwordLabel"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="password"
        >
          Password
        </label>
        <span>
          **********
        </span>
      </div>
    </div>
    <div
      className="changePasswordButtonContainer"
    >
      <button
        className="root_normalPriority"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="content"
        >
          Change Password
        </span>
      </button>
    </div>
    <div
      className="buttons"
    >
      <button
        className="root_lowPriority"
        disabled={false}
        onClick={[MockFunction onCancelModal]}
        type="button"
      >
        <span
          className="content"
        >
          Cancel
        </span>
      </button>
      <button
        className="root_highPriority"
        disabled={false}
        type="submit"
      >
        <span
          className="content"
        >
          Save
        </span>
      </button>
    </div>
  </form>,
]
`;

exports[`renders prefilled form with data with disabled buttons 1`] = `
Array [
  <FormError
    errors={Map {}}
  />,
  <form
    className="root"
    onKeyDown={[Function]}
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="firstname"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="firstname"
        >
          First Name
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              name="firstname"
              onBlur={[Function]}
              onChange={[Function]}
              value="Philip"
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="lastname"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="lastname"
        >
          Last Name
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              name="lastname"
              onBlur={[Function]}
              onChange={[Function]}
              value="Fry"
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="email"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="email"
        >
          Email
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              value="fry@planet.express"
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="passwordLabel"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="password"
        >
          Password
        </label>
        <span>
          **********
        </span>
      </div>
    </div>
    <div
      className="changePasswordButtonContainer"
    >
      <button
        className="root_normalPriority"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="content"
        >
          Change Password
        </span>
      </button>
    </div>
    <div
      className="buttons"
    >
      <button
        className="root_lowPriority"
        disabled={true}
        onClick={[MockFunction onCancelModal]}
        type="button"
      >
        <span
          className="content"
        >
          Cancel
        </span>
      </button>
      <button
        className="root_highPriority"
        disabled={true}
        type="submit"
      >
        <span
          className="content"
        >
          Saving
        </span>
      </button>
    </div>
  </form>,
]
`;

exports[`renders prefilled form with data with enabled change password form 1`] = `
Array [
  <FormError
    errors={Map {}}
  />,
  <form
    className="root"
    onKeyDown={[Function]}
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="firstname"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="firstname"
        >
          First Name
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              name="firstname"
              onBlur={[Function]}
              onChange={[Function]}
              value="Philip"
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="lastname"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="lastname"
        >
          Last Name
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              name="lastname"
              onBlur={[Function]}
              onChange={[Function]}
              value="Fry"
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="email"
    >
      <div
        className="root"
      >
        <label
          className="label"
          htmlFor="email"
        >
          Email
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 0,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              value="fry@planet.express"
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          />
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="password"
    >
      <div
        className="root"
      >
        <label
          className="label"
        >
          Current Password
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 1,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              autoComplete="current-password"
              className="input"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              type="password"
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          >
            <button
              className="passwordButton"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span
                className="content"
              >
                <svg
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M17.94 17.94A10.07 10.07 0 0 1 12 20c-7 0-11-8-11-8a18.45 18.45 0 0 1 5.06-5.94M9.9 4.24A9.12 9.12 0 0 1 12 4c7 0 11 8 11 8a18.5 18.5 0 0 1-2.16 3.19m-6.72-1.07a3 3 0 1 1-4.24-4.24"
                  />
                  <line
                    x1="1"
                    x2="23"
                    y1="1"
                    y2="23"
                  />
                </svg>
              </span>
            </button>
          </span>
        </span>
        <p
          className="root"
        />
      </div>
      <button
        className="root_normalPriority"
        disabled={false}
        type="button"
      >
        <span
          className="content"
        >
          Forgot Password?
        </span>
      </button>
    </div>
    <div
      className="newPassword"
    >
      <div
        className="root"
      >
        <label
          className="label"
        >
          New Password
        </label>
        <span
          className="root"
          style={
            Object {
              "--iconsAfter": 1,
              "--iconsBefore": 0,
            }
          }
        >
          <span
            className="input"
          >
            <input
              className="input"
              name="newPassword"
              onBlur={[Function]}
              onChange={[Function]}
              type="password"
              value=""
            />
          </span>
          <span
            className="before"
          />
          <span
            className="after"
          >
            <button
              className="passwordButton"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span
                className="content"
              >
                <svg
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M17.94 17.94A10.07 10.07 0 0 1 12 20c-7 0-11-8-11-8a18.45 18.45 0 0 1 5.06-5.94M9.9 4.24A9.12 9.12 0 0 1 12 4c7 0 11 8 11 8a18.5 18.5 0 0 1-2.16 3.19m-6.72-1.07a3 3 0 1 1-4.24-4.24"
                  />
                  <line
                    x1="1"
                    x2="23"
                    y1="1"
                    y2="23"
                  />
                </svg>
              </span>
            </button>
          </span>
        </span>
        <p
          className="root"
        />
      </div>
    </div>
    <div
      className="buttons"
    >
      <button
        className="root_lowPriority"
        disabled={false}
        onClick={[MockFunction onCancelModal]}
        type="button"
      >
        <span
          className="content"
        >
          Cancel
        </span>
      </button>
      <button
        className="root_highPriority"
        disabled={false}
        type="submit"
      >
        <span
          className="content"
        >
          Save
        </span>
      </button>
    </div>
  </form>,
]
`;
